### ISSUE TITLE
How can I use this eslint plugin without typescript?

### ISSUE THREAD

**Issue by zoedsoupe** [eslint-functional/eslint-plugin-functional#766]
## Bug Report

Tryingto to setup `extends` fields on `.eslintrc` as issued by the `GETTING_STARTED.md` file, raises an error that some rules depends on `@typescript/parser`

### Expected behavior

This plugin should work with raw javascript, am i right?

### Actual behavior

Raises this error on running `npx eslint .`:
```sh
npm run lint

> astrolog@1.0.0 lint
> npx eslint .


Oops! Something went wrong! :(

ESLint: 8.56.0

Error: You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Note: detected a parser other than @typescript-eslint/parser. Make sure the parser is configured to forward "parserOptions.project" to @typescript-eslint/parser.
Occurred while linting /Users/zoedsoupe/dev/andromeda/astrolog/src/main.js:4
Rule: "functional/no-return-void"
    at throwError (/Users/zoedsoupe/dev/andromeda/astrolog/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:39:11)
    at Object.getParserServices (/Users/zoedsoupe/dev/andromeda/astrolog/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:23:9)
    at getTypeOfNode (/Users/zoedsoupe/dev/andromeda/astrolog/node_modules/eslint-plugin-functional/lib/index.js:765:40)
    at checkFunction$2 (/Users/zoedsoupe/dev/andromeda/astrolog/node_modules/eslint-plugin-functional/lib/index.js:2061:34)
    at /Users/zoedsoupe/dev/andromeda/astrolog/node_modules/eslint-plugin-functional/lib/index.js:728:24
    at ruleErrorHandler (/Users/zoedsoupe/dev/andromeda/astrolog/node_modules/eslint/lib/linter/linter.js:1076:28)
    at /Users/zoedsoupe/dev/andromeda/astrolog/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/zoedsoupe/dev/andromeda/astrolog/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/zoedsoupe/dev/andromeda/astrolog/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
```

### Steps to reproduce

- start a fresh node project
- install eslint via `npm init @eslint/config` (do not use typescript)
- install this plugin as issued on the `GETTING_STARTED` guide
- uses this simple config:
```json
{
  "env": {
    "es2021": true,
    "node": true
  },
  "extends": [
    "prettier",
    "eslint:recommended",
    "plugin:functional/external-vanilla-recommended",
    "plugin:functional/recommended",
    "plugin:functional/stylistic"
  ],
  "parserOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "plugins": [
    "functional"
  ],
  "rules": {
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ]
  }
}
```
- try to run `npx eslint .`

### Proposed changes

I don’t know if I missed something on docs saying about it requires typescript, or even if I set up incorrectly...

**Comment by RebeccaStevens** [comment]
You'll need to disable the rules that require TypeScript to work.

```json
"rules": {
  "functional/immutable-data": "off",
  "functional/no-return-void": "off",
}
```
